File: rust_part/bubble_sort.rs
use std::convert::TryInto;


fn bubble_sort(mut arr: Vec<i32>)->Vec<i32>{
    let mut sorted: bool = false;
    let mut i: i32 = 0;
    let mut temp: i32;
    let length: i32 = (arr.len()).try_into().unwrap();
    let mut i_index: usize;
    let mut j_index: usize;
    while !(sorted){
        sorted = true;
        i = 0;
        while i<length-1{
            i_index = (i).try_into().unwrap();
            j_index = (i+1).try_into().unwrap();
            if arr[i_index] > arr[j_index]{
                sorted = false;
                temp = arr[i_index];
                arr[i_index] = arr[j_index];
                arr[j_index] = temp;
            }
            i += 1;
        }
    }


    return arr;
}


fn main(){
    let arr: Vec<i32> = vec![2, 1, 4, 13, 11, 9];
    let mut res: Vec<i32> = Vec::new();
    res = bubble_sort(arr.clone());
    let length: i32 = (arr.len()).try_into().unwrap();
    let mut index: usize;
    let mut i: i32 = 0;
    while i<length{
        index = (i).try_into().unwrap();
        println!("{}", res[index]);
        i += 1;
    }
}

File: c_part/bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

int* sorted(int *arr, const int size){
    int sorted = 0, i, temp;
    while(sorted == 0){
        sorted = 1;
        for(i=1; i<size; i++){
            if(arr[i] < arr[i-1]){
                sorted = 0;
                temp = arr[i];
                arr[i] = arr[i-1];
                arr[i-1] = temp;
            }
        }
    }

    return arr;
}


int main(){
    int arr[5] = {3, 2, 7, 1, 9};
    int* res = (int*)malloc(5*sizeof(int));
    res = sorted(arr, 5);

    for(int i=0; i<5; i++){
        printf("%d ", res[i]);
    }
    printf("\n");
}